src/main.ts

```diff
@@ -16,7 +16,9 @@ async function run(): Promise<void> {
     const registry = docker.getRegistry(inputs.repository);
     await docker.login(registry, inputs.username, inputs.password);
     await buildx.setup();
-    await docker.build(inputs);
+    if (!inputs.authOnly) {
+      await docker.build(inputs);
+    }
```

Comment:
No need to setup buildx container either I would just put this just after login:
```typescript
if (inputs.authOnly) {
  return
}
```